src/linter/ui5Types/fix/EventHandlersFix.ts

```diff
@@ -9,20 +9,19 @@ import SourceFileMetadataCollector from "../SourceFileMetadataCollector.js";
 export default class EventHandlersFix extends XmlEnabledFix {
 	protected sourcePosition: PositionInfo | undefined;
 	protected startPos: number | undefined;
-	protected endPos: number | undefined;
 	protected trailingCommaPos: number | undefined;
-	private fullMethodSignature: string | undefined;
-	private hasAvailableController = false;
+	private isMethodInController = false;
 
 	constructor(
 		private methodName: string,
-		private controllerName?: string
+		private controllerName: string,
+		sourcePosition: PositionInfo
 	) {
 		super();
+		this.sourcePosition = sourcePosition;
 	}
 
-	visitLinterNode(_node: ts.Node | AttributeDeclaration, sourcePosition: PositionInfo) {
-		this.sourcePosition = sourcePosition;
+	visitLinterNode(_node: ts.Node | AttributeDeclaration, _sourcePosition: PositionInfo) {
 		// If controller name is not present we cannot determine which is the actual controller, so skip further checks
```

Comment:
As far as I can see the current concept (e.g. implemented in `SourceFileLinter`) calls `visitLinterNode` after creating the fix (same as it has been done before the latest changes).

Should we continue with that, even though the function just needs to assign `sourcePosition` as member and return `true`?

I just think that now adding the `sourcePosition: PositionInfo` to the constructor seems odd, as it's against the pattern of all other fixes, isn't it?

But overall I think we should streamline the Fix classes to remove unrelated functionality, like having "ts" types in non-TS fixes.

Reply:
Yes, that's a good solution!